### PR TITLE
ci: add simple smoketest to binaries for docker, and add docker build to the PR ci

### PR DESCRIPTION
**Pull request by arirubinstein**
Adds a regression test for 
ref: [#7188](https://github.com/Agoric/agoric-sdk/issues/7188)
ref: [#6600](https://github.com/Agoric/agoric-sdk/issues/6600)

Includes a same-platform build for the sdk image

### CODE REVIEW

File: scripts/ci-collect-testruns.sh [hunk 1]
```diff
@@ -4,8 +4,10 @@ SCRIPT_DIR=$( cd ${0%/*} && pwd -P )
 
 for pkg in ./packages/*/_testoutput.txt
 do
-    echo "processing $pkg"
-    junit=$(dirname "$pkg")/junit.xml
-    echo "dest $junit"
-    $SCRIPT_DIR/ava-etap-to-junit.mjs $pkg > $junit
+    if [[ $pkg != "./packages/*/_testoutput.txt" ]];    then
```
Comment (arirubinstein) [inline scripts/ci-collect-testruns.sh]: lazy to handle non-expansion if there are no artifacts, in order to not fail the CI

File: .github/workflows/test-all-packages.yml [hunk 2]
```diff
@@ -466,3 +466,17 @@ jobs:
           if [ "$DATADOG_API_KEY" != "" ]; then
             ./scripts/ci-collect-testruns.sh && npx @datadog/datadog-ci junit upload --service agoric-sdk ./packages/*/junit.xml
           fi
+
+  test-docker-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: docker build (sdk)
+        timeout-minutes: 30
```
Comment (turadg) [inline .github/workflows/test-all-packages.yml]: seems high. what's the normal run time?

File: packages/deployment/scripts/smoketest-binaries.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+# For use inside of an environment (e.g. docker) to validate path of expected binaries work, and the binaries exit cleanly as well
+
+binaries=( "agd version" "agoric -V" "ag-chain-cosmos version" )
+for i in "${binaries[@]}"; do
```
Comment (turadg) [inline packages/deployment/scripts/smoketest-binaries.sh]: 'i' sounds like an index and this is a value. consider 'binary'

File: packages/deployment/scripts/test-docker-build.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+
+nowInMs() {
+  NO_COLOR=1 node -e "console.log(Date.now())"
+}
+
```
Comment (turadg) [inline packages/deployment/scripts/test-docker-build.sh]: Bash has this built in: https://stackoverflow.com/a/67044674/46040 Please use that.
```suggestion
```

File: packages/deployment/scripts/test-docker-build.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ec=$?
+sleep 1
+end=$(nowInMs)
+dur=$((end-start))
+
+echo "TAP version 13"
+echo "1..1"
+if [[ $ec -ne 0 ]]; then
+    fail=1
+    echo "not ok 1 - Docker Build %ava-dur=${dur}ms"
```
Comment (turadg) [inline packages/deployment/scripts/test-docker-build.sh:16]: `ava-dur` 😆

File: scripts/ci-collect-testruns.sh (hunk 1 above)
Comment (turadg) [reply scripts/ci-collect-testruns.sh]: important comment to have in the code.

File: .github/workflows/test-all-packages.yml (hunk 2 above)
Comment (arirubinstein) [reply .github/workflows/test-all-packages.yml]: ~12 minutes, however can be up to 30 if network issues are present

**Review by turadg — APPROVED** [review]
🚢

**Pull request merged**